src/Microsoft.Extensions.Localization/ResourceManagerStringLocalizerFactory.cs

```diff
@@ -62,7 +62,7 @@ protected virtual string GetResourcePrefix(TypeInfo typeInfo)
                 throw new ArgumentNullException(nameof(typeInfo));
             }
 
-            return GetResourcePrefix(typeInfo, _hostingEnvironment.ApplicationName, _resourcesRelativePath);
+            return GetResourcePrefix(typeInfo, new AssemblyName(typeInfo.Assembly.FullName).Name, _resourcesRelativePath);
```

Comment:
This change won't matter (`GetResourcePrefix()` will still return `typeInfo.FullName` which is a strange resource prefix) when `_resourcesRelativePath` is `null` or empty. If that's a possible scenario, should think the semantics through a bit more. If it's not possible, remove the ternary at [line 91](https://github.com/aspnet/Localization/pull/283/files#diff-ec3f46a4394df7156f745d976f053127R91).